### ISSUE TITLE
Dimmer - It is not possible to set level to 0

### ISSUE THREAD

**Issue by jelinekpetr** [domoticz/domoticz#5844]
Hi, I want to maybe only ask about dimmer function. 

First of all: I'm using docker version on Raspberry Pi. And now what's been happening: In version 2022.1 (I believe, not sure, I'm upgrading once a year) my LUA scripts were able to set level of virtual (dummy) dimmers to 0%. Something happened in the meanwhile and now I'm not able to set any dimmer to 0%. It is possible to set it to any other level but if I try to set it to 0%, dimmer is switched off and the level remains the same. The behavior is very similar if I try to use web GUI. When the slider touches the left edge, the dimmer goes to OFF state and level remains somewhere above zero. If the slider moves quickly the level stays somewhere between the original value (before moving)  and 1%. Never zero. If I move with slider very slowly, the last level is 1%. It seems like any attempt to dim to 0 causes only switching off but the level stays the last state known just before calling method dimTo(0) or touching slider very left edge.

I've found this in the last stable even beta versions.

My question is: is this behavior intended? I tried to find something abut this "issue" but with no success. For me it is not bad if the slider switches off when dimming to 0, it is fairly natural, but it is not perfectly logic if the level never can be 0%.

Thanks for any answer.

Petr

**Comment by gizmocuz** [comment]
That is correct, when the level to be set is received as 0, it is turned OFF.
When you turn it back ON again, it turns it on with the previous level
It is indeed intended. 0=OFF

**Comment by jelinekpetr** [comment]
Thanks for the answer. As I wrote it perfectly makes sense that the setting to zero making the dimmer switched off. No problem so far. But the fact the dimmer will never can be set to 0 is less understandable for me. I used the dimmers to control radiator valves and this is no longer possible that way I did it in the past. My scripts read levels of the dimmers to setting up the valves and the valves do not have any OFF state. So I will have to rewrite my scripts to check even the state of the dimmer (moreover to level) and translate OFF state to level 0. OK, no big deal because I don't use switchOff() and switchOn() methods and state == 'OFF' is for me simply 0. But I can imagine applications where both switching methods are used and the zero level is normal value. Then you cannot easily determine whether the OFF state is result of method switchOff() or method dimTo(0).

So thanks again for the explanation and I will try to get used to it. :-)

Petr
